autocomplete/builders/common.lua

```diff
@@ -393,54 +408,102 @@ end
 -- Package compilation
 --
 
+---@alias packageType
+---|"class"
+---|"function"
+---|"method"
+---|"value"
+---|"lib"
+---|"event"
+
 --- @class exampleTable
 --- @field title string|nil The example title.
 --- @field description string|nil The description of the example.
 
 --- @class package
+--- @field name string
+--- @field useDefault boolean
+--- @field readOnly boolean
+--- @field optional boolean? Is this parameter optional?
+--- @field default string|integer|nil
+--- @field experimental boolean
+--- @field description string?
 --- @field key string The name of the file that generated this package.
---- @field type string The type definition for the package.
+--- @field type packageType The type definition for the package.
 --- @field folder string The folder that the package was created from.
 --- @field parent package The package this package is a child of.
 --- @field namespace string The full namespace of the package.
 --- @field deprecated boolean Allows marking definitions as deprecated. Those definitions aren't written to the web documentation.
 --- @field examples table<string, exampleTable>|nil A table containing the examples. Keys are the example's name/path to the example file.
 
+---@class packageValue : package
```

Comment:
Can you add a description to this class to link to https://github.com/MWSE/MWSE/blob/master/docs/type-definitions-guide.md#value-definitions. Can also be made to link to the local file: [link text](../../docs/type-definitions-guide.md)